packages/model-viewer/src/three-components/SmoothControls.ts

```diff
@@ -839,33 +864,168 @@ export class SmoothControls extends EventDispatcher {
+        this.userAdjustOrbit(0, -OrbitInc, 0);
         break;
       case KeyCode.DOWN:
-        this.userAdjustOrbit(0, KEYBOARD_ORBIT_INCREMENT, 0);
+        this.userAdjustOrbit(0, OrbitInc, 0);
         break;
       case KeyCode.LEFT:
-        this.userAdjustOrbit(-KEYBOARD_ORBIT_INCREMENT, 0, 0);
+        this.userAdjustOrbit(-OrbitInc, 0, 0);
         break;
       case KeyCode.RIGHT:
-        this.userAdjustOrbit(KEYBOARD_ORBIT_INCREMENT, 0, 0);
+        this.userAdjustOrbit(OrbitInc, 0, 0);
+        break;
+      case KeyCode.KEY_1:
+      case KeyCode.NP_KEY_1:
+          this.keyPressThetaSet(0);
+        break;
+      case KeyCode.KEY_2:
+      case KeyCode.NP_KEY_2:
+          this.keyPressThetaSet(Math.PI /2);
         break;
+      case KeyCode.KEY_3:
+      case KeyCode.NP_KEY_3:        
+          this.keyPressThetaSet(Math.PI);
+        break;
+      case KeyCode.KEY_4:
+      case KeyCode.NP_KEY_4:
+          this.keyPressThetaSet(Math.PI * 1.5);
+        break;
+      case KeyCode.KEY_7:
+      case KeyCode.NP_KEY_7:
+          this.keyPressPhiSet(Math.PI *3 /4);
+        break;
+      case KeyCode.KEY_8:
+      case KeyCode.NP_KEY_8:
+          this.keyPressPhiSet(Math.PI /2);
+        break;
+      case KeyCode.KEY_9:
+      case KeyCode.NP_KEY_9:
+          this.keyPressPhiSet(Math.PI / 4);
+        break;   
       default:
-        relevantKey = false;
+        swizzkey = false;
         this.isUserChange = isUserChange;
         break;
     }
+    return swizzkey;
+  }
 
-    if (relevantKey) {
-      event.preventDefault();
+  private panKeyCodeHandler(event: KeyboardEvent, dx: number, dy:number) {
+    this.initializePan();
+    let swizzkey = true;
+    switch (event.keyCode) {
+      case KeyCode.PAGE_UP:
+        // Do nothing for this
+        // this.userAdjustOrbit(0, 0, ZoomInc);
+        break;
+      case KeyCode.PAGE_DOWN:
+        // Do nothing for this
+        // this.userAdjustOrbit(0, 0, -1 * ZoomInc);
+        break;
+      case KeyCode.UP:
+        this.movePan(0, -1 * dy); // This is the negative one so that the model appears to move as the arrow direction rather than the view moving
+        break;
+      case KeyCode.DOWN:
+        this.movePan(0, dy);
+        break;
+      case KeyCode.LEFT:
+        this.movePan(-1 * dx, 0);
+        break;
+      case KeyCode.RIGHT:
+        this.movePan(dx, 0);
+        break;
+      default:
+        swizzkey = false;
+        break;
     }
-  };
+    return swizzkey;
+  }
+
+  // Adjust only the theta of the orbit
+  // (Effectively spinning around the target horizontally)
+  private keyPressThetaSet(newTheta: number) {
+    const {phi, radius} = this.goalSpherical;
+    this.setOrbit(newTheta, phi, radius);
+  }
+
+  // Adjust only the phi of the orbit
+  // (Effectively spinning around the target vertically)
+  private keyPressPhiSet(newPhi: number) {
+    const {theta, radius} = this.goalSpherical;
+    this.setOrbit(theta, newPhi, radius);
+  }
 }
+
```

Comment:
Can you find a way to remove all those extra empty diff files? Not sure where those are coming from.

Reply:
Looks like this? https://github.com/desktop/desktop/issues/557 The end of the thread has some things to try.

Reply:
This has got to be something to do with it. Investigation under way.

Reply:
This has got to be something to do with it. Investigation under way.